src/app/screens/Nostr/ConfirmSignMessage.tsx

```diff
@@ -0,0 +1,118 @@
+//import Checkbox from "../../components/Form/Checkbox";
+import ConfirmOrCancel from "@components/ConfirmOrCancel";
+import Container from "@components/Container";
+import ContentMessage from "@components/ContentMessage";
+import PublisherCard from "@components/PublisherCard";
+import SuccessMessage from "@components/SuccessMessage";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import ScreenHeader from "~/app/components/ScreenHeader";
+import { useNavigationState } from "~/app/hooks/useNavigationState";
+import { USER_REJECTED_ERROR } from "~/common/constants";
+import msg from "~/common/lib/msg";
+import type { OriginData } from "~/types";
+
+function ConfirmSignMessage() {
+  const navState = useNavigationState();
+  const { t: tCommon } = useTranslation("common");
+  const { t } = useTranslation("translation", {
+    keyPrefix: "confirm_sign_message",
+  });
+  const navigate = useNavigate();
+
+  const message = navState.args?.message as string;
+  const origin = navState.origin as OriginData;
+  //const [rememberMe, setRememberMe] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [successMessage, setSuccessMessage] = useState("");
+
+  // TODO: refactor: the success message and loading will not be displayed because after the reply the prompt is closed.
+  async function confirm() {
+    try {
+      setLoading(true);
+      msg.reply({
```

Comment:
following the other/newer conventions we should do the signing call here and return the response then back.

Reply:
really? if we move to more generic prompts & permissions we would need to move those things out again, no?

Reply:
i feel like those prompts are more like javascript confirms: 

```
if(confirm("do you really want to do x?")) {
  // do x
}
```

Reply:
I don't know what's the best approach here. 

yes, it is like `confirm()` but much more complexity can be behind that. 
one approach is to simply return true/false (which works for those simple confirms) 

or to do the actual call in the prompt and then return the response of that one. 

In here I do it also like in this PR here: https://github.com/getAlby/lightning-browser-extension/pull/1707 (simple return true/false and have the logic in the calling action)